spec/models/room_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+RSpec.describe Room, type: :model do
+  context "Factory" do
+    it{is_expected.to have_db_column(:name).of_type(:string)}
+    it{is_expected.to have_db_column(:occupancy_limit).of_type(:integer)}
+    it{is_expected.to have_db_column(:living_room).of_type(:integer)}
+    it{is_expected.to have_db_column(:bed_room).of_type(:integer)}
+    it{is_expected.to have_db_column(:bath_room).of_type(:integer)}
+    it{is_expected.to have_db_column(:number_of_bed).of_type(:integer)}
+    it{is_expected.to have_db_column(:quantity).of_type(:integer)}
+    it{is_expected.to have_db_column(:price).of_type(:decimal)}
+    it{is_expected.to have_db_column(:status).of_type(:boolean)}
+  end
+
+  context "Validations" do
+    before do
+      subject{build(:room)}
```

Comment:
subject dùng để làm j

Reply:
Subject là  đối tướng đang được test .Nó có thể có hoặc không được xác định . Nếu được xác định Rspec có thể gọi các phương thức mà k cần đề cập đến nó.
-Implicitly subject: Khi argument đầu tiên của example group ngoài cùng là một class, thì một instance của class đó sẽ được truyền đến mỗi example thông qua subject .
VD: expect(subject).to be_empty
-Explicit subject: xác định 1 giá trị rõ ràng được trả về bởi subject method trong 1 example
VD : subject { [1, 2, 3] }